### PR TITLE
Add comments

### PR DESCRIPTION
**Pull request by LacticWhale**

### CODE REVIEW

File: lib/src/API.dart [hunk 1]
```diff
@@ -80,6 +80,13 @@ class Hosts {
     ),
     '/galleries/${image.book.media}/${image.filename}',
   );
+
+
+  /// Get user avatar url with respect to hosts settings.
+  Uri getAvatarUrl(String avatarUrl) => Uri.https(
```
Comment (Zekfad) [inline lib/src/API.dart]: This function must accept `User` not a `String`.

File: lib/src/API.dart [hunk 2]
```diff
@@ -184,6 +191,17 @@ class API {
   /// Get [image] url with respect to this client instance [hosts] settings.
   Uri getImageUrl(Image image) => hosts.getImageUrl(image);
 
+  /// Get comments associated with given book ID.
+  Future<List<Comment>?> getComments(int bookId) async {
+    assert(bookId > 0, 'Id must be positive integer.');
+    return tryParseList<Comment>(
+      _getJson(_getPath(HostType.api, '/api/gallery/$bookId/comments')),
```
Comment (Zekfad) [inline lib/src/API.dart]: Place `_getPath` on a new line, please.

File: lib/src/API.dart [hunk 3]
```diff
@@ -184,6 +191,17 @@ class API {
...
+    );
+  }
+
+  /// Get user avatar url with respect to this client instance [hosts] settings.
+  Uri getAvatarUrl(String avatarUrl) => hosts.getAvatarUrl(avatarUrl);
```
Comment (Zekfad) [inline lib/src/API.dart]: This function must accept `User` not a `String`.

Comment (Zekfad) [reply lib/src/API.dart]: Place this function after `getImageUrl`.

File: lib/src/User.dart [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  /// User name.
+  final String username;
+  /// User slug.
+  final String slug;
+  /// User relevant avatar path.
+  final String avatarUrl;
+  /// Is user a superuser.
+  final bool isSuperuser;
+  /// Is user a stuff.
+  final bool isStaff;
```
Comment (Zekfad) [inline lib/src/User.dart]: Drop `is`.

File: lib/src/Comment.dart [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    required this.postDate,
+    required this.body,
+  }) : super();
+
+  /// Comment ID.
+  final int id;
+  /// Gallery ID associated with comment.
+  final int galleryId;
+  /// Comment author.
+  final User poster;
```
Comment (Zekfad) [inline lib/src/Comment.dart]: Name it `user`.

File: lib/src/Comment.dart [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    required this.galleryId,
+    required this.poster,
+    required this.postDate,
+    required this.body,
+  }) : super();
+
+  /// Comment ID.
+  final int id;
+  /// Gallery ID associated with comment.
+  final int galleryId;
```
Comment (Zekfad) [inline lib/src/Comment.dart]: Name it `bookId`.

File: lib/src/Comment.dart [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import 'DataModel.dart';
+import 'parseUtils.dart' show tryParseAsTyped, parseAsTyped, parseList, parseMap;
+
+/// Comment.
+class Comment {
+  /// Create comment.
+  Comment({
```
Comment (Zekfad) [inline lib/src/Comment.dart]: This class is const.

File: lib/src/User.dart [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import 'parseUtils.dart' show parseAsTyped;
+
+/// User.
+class User {
+  /// Create user.
+  User({
```
Comment (Zekfad) [inline lib/src/User.dart]: This class is const.

**Review by Zekfad — CHANGES_REQUESTED** [review]
Whole `avatar url` stuff isn't needed.
They all follow same rules: `avatars/${userId}.png` or `avatars/pastamania.png`
Make it either `avatarFilename` or `boolean` `hasAvatar`, and add to `getAvatarUrl` named parameter `fallback` with default value `pastamania.png`.

> One is a general solution, the other is memory efficient.
> I guess it's better to stick with filename.



Also add `getAvatarUrl` to `User` like that: https://github.com/Zekfad/nhentai_dart/blob/5bb4fa77477af8b7963d1d610403ff8b1f655d43/lib/src/Image.dart#L106-L114

File: lib/src/API.dart [hunk 9]
```diff
@@ -83,9 +83,9 @@ class Hosts {
 
 
   /// Get user avatar url with respect to hosts settings.
-  Uri getAvatarUrl(String avatarUrl) => Uri.https(
+  Uri getAvatarUrl(User user) => Uri.https(
     getHost(HostType.images),
-    avatarUrl,
+    user.avatarFilename,
```
Comment (Zekfad) [inline lib/src/API.dart]: That's wrong.

**Review by Zekfad — CHANGES_REQUESTED** [review]

File: lib/src/API.dart [hunk 10]
```diff
@@ -169,6 +169,12 @@ class API {
     return response.headers[HttpHeaders.locationHeader]!.first;
   }
 
+  /// Get [image] url with respect to this client instance [hosts] settings.
+  Uri getImageUrl(Image image) => hosts.getImageUrl(image);
+
+  /// Get user avatar url with respect to this client instance [hosts] settings.
```
Comment (Zekfad) [inline lib/src/API.dart]: Mark `user` as `[user]` in docs.

Comment (Zekfad) [inline lib/src/API.dart]: Change to "with respect to [hosts] settings", please.

File: lib/src/User.dart [hunk 11]
```diff
@@ -25,6 +26,16 @@ class User {
   /// Is user a stuff.
```
Comment (Zekfad) [inline lib/src/User.dart]: Change to "Whether user is staff member", please.

File: lib/src/User.dart [hunk 12]
```diff
@@ -25,6 +26,16 @@ class User {
...
   final bool staff;
 
+  Uri getAvatarUrl({
```
Comment (Zekfad) [inline lib/src/User.dart:30]: Add docs.
Example:
https://github.com/Zekfad/nhentai_dart/blob/dba010658c2b965f77d91e4e501d58edb654db7c/lib/src/Image.dart#L105-L109

File: lib/src/User.dart [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+    return url;
+  }
+
+  /// Get [username].
+  @override
+  String toString() => username;
+
+  /// Parse user from API [json] object.
+  static User? parse(dynamic json) {
+    if(json == null)
```
Comment (Zekfad) [inline lib/src/User.dart]: Space after `if`.

File: pubspec.yaml [hunk 14]
```diff
@@ -10,6 +10,7 @@ environment:
   sdk: '>=2.16.0 <3.0.0'
 
 dev_dependencies:
+  http: ^0.13.4
```
Comment (Zekfad) [inline pubspec.yaml]: ?

File: lib/src/API.dart [hunk 15]
```diff
@@ -80,6 +80,13 @@ class Hosts {
     ),
     '/galleries/${image.book.media}/${image.filename}',
   );
+
+
```
Comment (Zekfad) [inline lib/src/API.dart]: Remove 1 newline.

File: lib/src/Comment.dart [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+  /// Comment content.
+  final String body;
+
+  /// Get [body].
+  @override
+  String toString() => body;
+
+  /// Parse comment from API [json] object.
+  static Comment? parse(dynamic json) {
+    if(json == null)
```
Comment (Zekfad) [inline lib/src/Comment.dart]: Space after `if`.

File: lib/src/Comment.dart [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+  /// Gallery ID associated with comment.
+  final int bookId;
+  /// Comment author.
+  final User author;
+  /// Comment post date.
+  final DateTime date;
+  /// Comment content.
+  final String body;
+
+  /// Get [body].
```
Comment (Zekfad) [inline lib/src/Comment.dart]: "Get comment [body]".

File: lib/src/User.dart [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+    required this.avatarFilename,
+    required this.superuser,
+    required this.staff,
+  }) : super();
+
+  /// User ID.
+  final int id;
+  /// User name.
+  final String username;
+  /// User slug.
```
Comment (Zekfad) [inline lib/src/User.dart]: "User slug (unique short username for URLs)"

File: lib/src/User.dart [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+  final String slug;
+  /// User relevant avatar path.
+  final String avatarFilename;
+  /// Is user a superuser.
```
Comment (Zekfad) [inline lib/src/User.dart]: "Whether user is superuser"

File: lib/src/User.dart [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+    required this.staff,
+  }) : super();
+
+  /// User ID.
+  final int id;
+  /// User name.
+  final String username;
+  /// User slug.
+  final String slug;
+  /// User relevant avatar path.
```
Comment (Zekfad) [inline lib/src/User.dart]: "User's avatar file name"

File: lib/src/API.dart [hunk 21]
```diff
@@ -181,8 +194,15 @@ class API {
     );
   }
 
-  /// Get [image] url with respect to this client instance [hosts] settings.
-  Uri getImageUrl(Image image) => hosts.getImageUrl(image);
+  /// Get comments associated with given book ID.
```
Comment (Zekfad) [inline lib/src/API.dart]: "Get book comments"

File: lib/src/Comment.dart [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+    required this.bookId,
+    required this.author,
+    required this.date,
+    required this.body,
+  }) : super();
+
+  /// Comment ID.
+  final int id;
+  /// Gallery ID associated with comment.
+  final int bookId;
```
Comment (Zekfad) [inline lib/src/Comment.dart]: "Gallery" -> "Book"

**Review by Zekfad — CHANGES_REQUESTED** [review]
Some changes required.

File: pubspec.yaml (hunk 14 above)
Comment (LacticWhale) [reply pubspec.yaml]: bruh

File: lib/src/User.dart (hunk 13 above)
Comment (Zekfad) [reply lib/src/User.dart]: still

**Review by Zekfad — CHANGES_REQUESTED** [review]

**Review by Zekfad — APPROVED** [review]

**Pull request merged**